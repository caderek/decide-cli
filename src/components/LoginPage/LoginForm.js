import React, { Component } from 'react'
import $ from './LoginForm.scss'

class LoginForm extends Component {
  constructor (props) {
    super(props)
    this.state = {
      user: '',
      password: ''
    }

    this.handleChange = this.handleChange.bind(this)
    this.authenticate = this.authenticate.bind(this)
  }

  authenticate (event) {
    event.preventDefault()
    this.props.authenticate(this.state.user, this.state.password)
  }

  handleChange ({ target }) {
    this.setState({ [target.name]: target.value })
  }

  render () {
    return (
      <form className={$.loginForm} onSubmit={this.authenticate} >
        <input
          type='text'
          name='user'
          placeholder='user'
          value={this.state.user}
          onChange={this.handleChange}
          autoFocus
        />
        <input
          type='password'
          name='password'
          placeholder='password'
          value={this.state.password}
          onChange={this.handleChange}
        />
        <input type='submit' value='log in' />
      </form>
    )
  }
}

export default LoginForm
