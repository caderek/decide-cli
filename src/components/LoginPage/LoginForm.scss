.loginForm {
  max-width: 330px;
  margin: auto;

  input[type=text], input[type=password] {
    padding: 10px;
    font-size: 24px;
    font-family: "Fira Code";
    margin-bottom: 10px;
    width: 100%;
    border: solid 1px #C0C1C0;
    outline: none;

    &:focus {
      border-color: #F6A623;
    }
  }

  input[type=submit] {
    font-size: 24px;
    font-family: "Fira Code";
    background-color: #72889A;
    color: #F8F8F8;
    padding: 10px;
    width: 100%;
    outline: none;
    cursor: pointer;

    &:hover {
      background-color: #5B6D7D;
    }
  }
}

