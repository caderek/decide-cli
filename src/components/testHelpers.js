import renderer from 'react-test-renderer'

export const snap = component => {
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
}
