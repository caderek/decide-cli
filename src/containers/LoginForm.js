import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { authenticate } from '../actions'
import LoginForm from '../components/LoginPage/LoginForm'

const mapDispatch = (dispatch) => bindActionCreators({ authenticate }, dispatch)
export default connect(null, mapDispatch)(LoginForm)

