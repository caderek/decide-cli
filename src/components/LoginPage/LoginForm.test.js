import React from 'react'
import { snap } from '../testHelpers'
import LoginForm from './LoginForm'

test('LoginForm renders correctly', () => snap(
  <LoginForm />
))
