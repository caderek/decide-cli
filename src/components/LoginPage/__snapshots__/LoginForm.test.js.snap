exports[`test LoginForm renders correctly 1`] = `
<form
  className="loginForm"
  onSubmit={[Function]}>
  <input
    autoFocus={true}
    name="user"
    onChange={[Function]}
    placeholder="user"
    type="text"
    value="" />
  <input
    name="password"
    onChange={[Function]}
    placeholder="password"
    type="password"
    value="" />
  <input
    type="submit"
    value="log in" />
</form>
`;
